frontend/apps/reader/modules/readerhighlight.lua

```diff
@@ -721,9 +721,13 @@ function ReaderHighlight:onHoldRelease()
                     },
                     {
                         text = _("Add Note"),
-                        enabled = false,
+                        enabled = true,
                         callback = function()
-                            self:addNote()
+                            local page, index = self:saveHighlight()
+                            if page and index then -- saveHighlight can return nil, nil
+                                self:editHighlight(page, index)
+                                UIManager:close(self.edit_highlight_dialog)
+                            end
```

Comment:
You might also want to add an `else` like:
```lua
UIManager:show(InfoMessage:new{
    text = _("Something about why saveHighlight can return nil, nil."),
    show_icon = false,
    timeout = 3,
})
```

Reply:
This comment is mostly for developers, this can only happen when text is not selected, e.g. 
```lua
if self.hold_pos and self.selected_text and self.selected_text.pos0 and self.selected_text.pos1 then
```
is false. I'm not sure what can cause this, so I left it in case this situation occurs.

Reply:
And this isn't something you can check in the enabled_func?

Reply:
This is already being checked in `ReaderHighlight:saveHighlight`, I don't think there is any point checking it again in enabled_func. In fact, I think that dialog won't be shown if this is false, although I haven't checked it yet. This is not checked for other buttons too.

Reply:
Then the question automatically becomes the reverse. ;-)

Reply:
I'm pretty sure in this context those additional checks were not necessary, so I've removed them.